### ISSUE TITLE
Service Name incorrectly encoded in DBM Trace Propagation

### ISSUE THREAD

**Issue by jcarvalho** [DataDog/dd-trace-js#2697]
Hi.

We're trying out the new APM DBM Link for Postgres feature that got introduced in 3.9.0 (https://github.com/DataDog/dd-trace-js/pull/2480), but noticed that the Service Name isn't being correctly encoded when a Service Name function is used.

**Expected behaviour**
The injected DBM query comment resolves the service name and injects it correctly. Example:

```sql
/*dddbs='DUMMY',dde='development',ddps='backend',ddpv='1.0.0',traceparent='01-000000000000000064186e7c969771f2-4c44621c4eedd5ca-00'*/
```

**Actual behaviour**
The a serialized form of the function is rendered:

```sql
/*dddbs='(params)%20%3D%3E%20%7B%0A%20%20%20%20%20%20%20%20%20%20%20%20return%20%22dummy%22.toUpperCase()%3B%0A%20%20%20%20%20%20%20%20%7D',dde='development',ddps='backend',ddpv='1.0.0',traceparent='01-000000000000000064186e7c969771f2-4c44621c4eedd5ca-00'*/
```

**Steps to reproduce**
Configure the Service Name for the `pg` plugin using a function (using a dummy function here for illustrative purposes):

```typescript
  tracer.use("pg", {
    service: (params: any) => {
      return "dummy".toUpperCase();
    },
  });
```

And then, enable DBM propagation using:

```typescript
    dbmPropagationMode: "full",
```

I've traced the issue down to [this line](https://github.com/DataDog/dd-trace-js/blob/v3.11.0/packages/dd-trace/src/plugins/database.js#L29), where the `this.config.service` property is being read directly, and not invoking the function. I don't think that the service name will be fully resolvable at the `DatabasePlugin` level, so it may need to be resolved directly in the PG plugin and passed as an argument to `injectDbmQuery` (happy to open a PR to fix this).

**Environment**

* **Operation system:** Linux
* **Node.js version:** 18.13.0
* **Tracer version:** 3.11.0
* **Agent version:** N/A
* **Relevant library versions:** `pg` @ 8.8.0

**Comment by tlhunter** [comment]
This should be fixed in the next release.

**Comment by jcarvalho** [comment]
Awesome, thanks! I'll keep an eye on the release and report back

**Comment by tlhunter** [comment]
This should be available in v3.12.0 and v2.25.0.
